common/src/main/java/com/microsoft/identity/common/internal/request/AuthenticationSchemeTypeAdapter.java

```diff
@@ -32,25 +34,35 @@
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
 import com.microsoft.identity.common.java.authscheme.AbstractAuthenticationScheme;
-import com.microsoft.identity.common.internal.authscheme.BearerAuthenticationSchemeInternal;
-import com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal;
+import com.microsoft.identity.common.java.authscheme.BearerAuthenticationSchemeInternal;
+import com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal;
 import com.microsoft.identity.common.logging.Logger;
 
 import java.lang.reflect.Type;
 
 import static com.microsoft.identity.common.java.authscheme.AbstractAuthenticationScheme.SerializedNames.NAME;
-import static com.microsoft.identity.common.internal.authscheme.BearerAuthenticationSchemeInternal.SCHEME_BEARER;
-import static com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal.SCHEME_POP;
+import static com.microsoft.identity.common.java.authscheme.BearerAuthenticationSchemeInternal.SCHEME_BEARER;
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SCHEME_POP;
 
 /**
  * Gson de/serialization utility class for auth schemes.
  */
-class AuthenticationSchemeTypeAdapter implements
+public class AuthenticationSchemeTypeAdapter implements
         JsonDeserializer<AbstractAuthenticationScheme>,
         JsonSerializer<AbstractAuthenticationScheme> {
 
     private static final String TAG = AuthenticationSchemeTypeAdapter.class.getSimpleName();
 
+    private static Gson sRequestAdapterGsonInstance = new GsonBuilder()
```

Comment:
(thought, not to be dealt with now) When I've done this with XML using JaxbContext, it helped to have one global JaxbContext for the entire application with all types defined therein.  I'm sure that the Json equivalent did something similar when it came time to shift serialization formats. I'm wondering if a similar pattern should hold for us.